tests/garage/tf/policies/test_deterministic_mlp_policy_with_model.py

```diff
@@ -0,0 +1,66 @@
+import pickle
+from unittest import mock
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.envs import TfEnv
+from garage.tf.models import Model
+from garage.tf.policies import DeterministicMLPPolicyWithModel
+from tests.fixtures import TfGraphTestCase
+from tests.fixtures.envs.dummy import DummyBoxEnv
+
+
+class SimpleMLPModel(Model):
+    """Simple MLPModel for testing."""
+
+    def __init__(self, name, output_dim, *args, **kwargs):
+        super().__init__(name)
+        self.output_dim = output_dim
+
+    def _build(self, obs_input):
+        return tf.constant(
+            1., shape=(
+                1,
+                self.output_dim,
+            ))
+
+
+class TestDeterministicMLPPolicyWithModel(TfGraphTestCase):
+    def setUp(self):
+        super().setUp()
+        self.env = TfEnv(DummyBoxEnv())
+        self.env.reset()
+        self.obs, _, _, _ = self.env.step(1)
+        model_path = 'garage.tf.policies.' + \
+                     'deterministic_mlp_policy_with_model.MLPModel'
+        with mock.patch(model_path, new=SimpleMLPModel):
+            self.policy = DeterministicMLPPolicyWithModel(
+                env_spec=self.env.spec)
+
+    def test_get_action(self):
+        action, _ = self.policy.get_action(self.obs)
+        assert self.env.action_space.contains(np.asarray([action]))
```

Comment:
why is np.asarray required here? why is get_action returning something that needs to be converted before being used with a space?

Reply:
I think it's actually an issue.
e.g. In akro.box, the `contains` function is as following:
```python
    def contains(self, x):
        """Return boolean specifying if x is a valid member of this space."""
        return x.shape == self.shape and (x >= self.low).all() and (
            x <= self.high).all()
```
It assumes the input is a numpy array. 
Because the return action from `policy.get_action` can be a scalar value, it won't work.
Should we force the output of `policy.get_action` to be a numpy array?

Reply:
it looks like you found a `n = 1` vs `n > 1` bug. good job!

i think that the output of get_action should always pass the test
```python
a = foo.get_action(bar)
assert a in foo.action_space

bs = foo.get_actions(bars)
for b in bs:
    assert b in foo.action_space
```

the question is -- is this a garage bug or an akro bug?

i think it is a garage bug. the output type of a policy should be fixed -- it should't be "a scalar sometimes and an np.ndarray other times." if the value is single-dimensional, i think it should output an ndarray with shape `(1,)`.

of course, i could be persuaded otherwise if this made code elsewhere really nasty.